packages/optivum-scrapper/src/timetable.ts

```diff
@@ -38,43 +38,54 @@ export class Timetable {
         return new Table(response);
     }
 
-    public async getUnitList(): Promise<Unit[]> {
-        const { response, responseUrl } = await this.getDocument(this.baseUrl);
-        this.baseUrl = responseUrl;
+    private async getIndexPageFromScript() {
+        const scriptResponse = (await this.getDocument('../scripts/powrot.js')).response;
+        const { response, responseUrl } = await this.getDocument(
+            /<a href="(.*?)">Plan lekcji<\/a>/.exec(scriptResponse)?.[1] ?? '../index.html',
+        );
+        return [response, responseUrl];
+    }
+
+    private async getListPageFromFrame(frame: Element) {
+        const listPath = frame.getAttribute('src')!;
+        const { response, responseUrl } = await this.getDocument(listPath);
+        return [response, responseUrl];
+    }
+
+    public async getUnitList(): Promise<{ units: Unit[]; sources: string[] }> {
+        const { response } = await this.getDocument(this.baseUrl);
         let document = new JSDOM(response).window.document;
         if (document.querySelector('script[src="../scripts/powrot.js"]') !== null) {
-            const scriptResponse = (await this.getDocument('../scripts/powrot.js')).response;
-            const { response, responseUrl } = await this.getDocument(
-                /<a href="(.*?)">Plan lekcji<\/a>/.exec(scriptResponse)?.[0] ?? '../index.html',
-            );
+            const [response, responseUrl] = await this.getIndexPageFromScript();
             this.baseUrl = responseUrl;
             document = new JSDOM(response).window.document;
         }
         if (document.querySelector('.menu') !== null) {
-            const list: Unit[] = [];
+            const list: { units: Unit[]; sources: string[] } = { units: [], sources: [] };
             await Promise.all(
-                [...document.querySelectorAll('a[hidefocus="true"]')].map(async (listLink) => {
-                    const href = listLink.getAttribute('href');
-                    if (href == null) return;
-                    const { response } = await this.getDocument(href);
+                [...document.querySelectorAll('a[hidefocus="true"][href]')].map(async (listLink) => {
+                    const href = listLink.getAttribute('href')!;
+                    const { response, responseUrl } = await this.getDocument(href);
                     const unitList = this.parseUnitList(response);
-                    list.push(...unitList);
+                    list.units.push(...unitList);
+                    list.sources.push(responseUrl);
                 }),
             );
+            list.sources = list.sources.sort();
```

Comment:
`.sort()` działa in-place, wystarczy coś takiego:
```suggestion
            list.sources.sort();
```